hyperspy/io_plugins/emd.py

```diff
@@ -681,7 +681,11 @@ def _read_image(self, image_group, image_sub_group_key):
                     chunks=h5data.chunks),
                 axes=[2, 0, 1])
         else:
-            data = np.rollaxis(np.array(h5data), axis=2)
+            # Preallocate the numpy array and use read_direct method, which is
+            # much faster in case of chunked data.
```

Comment:
Could you add a comment explaining that this is a workaround for a bug in h5py with a link to the relevant issue?

Reply:
Done!